pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,365 @@
+        Teardown: TeardownMessage(usage),
+        Fields: size_fields.concat(passphrase_fields),
+        Action: {
+            Title: _("Grow"),
+            action: function (vals) {
+                return teardown_active_usage(client, usage)
+                        .then(function () {
+                            return (lvol_and_fsys_resize(client, lvol,
+                                                         to_fit ? grow_size : vals.size,
+                                                         info.grow_needs_unmount,
+                                                         vals.passphrase || recovered_passphrase)
+                                    .catch(request_passphrase_on_error_handler(dlg, vals, recovered_passphrase, block)));
+                        });
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage),
+            passphrase_fields.length
+                ? init_existing_passphrase(block, false, pp => { recovered_passphrase = pp })
+                : null
+        ]
+    });
+}
+
+export function shrink_dialog(client, lvol, info, to_fit) {
+    const block = client.lvols_block[lvol.path];
+    const vgroup = client.vgroups[lvol.VolumeGroup];
+
+    const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
+                                   _("shrink"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let shrink_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: lvol.Size,
+                           max: lvol.Size,
+                           round: vgroup.ExtentSize
+                       })
+        ];
+    } else {
+        const crypto = client.blocks_crypto[block.path];
+        const cleartext = client.blocks_cleartext[block.path];
+        let content_path = null;
+        let crypto_overhead = 0;
+
+        if (crypto) {
+            if (crypto.MetadataSize !== undefined && cleartext) {
+                content_path = cleartext.path;
+                crypto_overhead = crypto.MetadataSize;
+            }
+        } else {
+            content_path = block.path;
+        }
+
+        const fsys = client.blocks_fsys[content_path];
+        if (fsys)
+            shrink_size = fsys.Size + crypto_overhead;
+
+        const vdo = client.legacy_vdo_overlay.find_by_backing_block(client.blocks[content_path]);
+        if (vdo)
+            shrink_size = vdo.physical_size + crypto_overhead;
+
+        const stratis_bdev = client.blocks_stratis_blockdev[content_path];
+        if (stratis_bdev)
+            shrink_size = Number(stratis_bdev.TotalPhysicalSize) + crypto_overhead;
+
+        if (shrink_size === undefined) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-121722-6ccb869e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)